_source/_posts/2022-01-20-vue-applications-nuxt.md

```diff
@@ -0,0 +1,357 @@
+npm i @nuxtjs/auth-next@5.0.0-1637333559.35dbd53
+```
+
+With your dependencies installed it's time to start building your application. First, add a new file to your root directory and name it `.env` and add the following code to it.
+
+```bash
+OKTA_DOMAIN=https://{yourOktaDomain}
+OKTA_CLIENT_ID={yourClientId}
+```
+
+Be sure to replace the placeholder variables with your actual Okta information.
+
+Now open your `nuxt.config.js` file located in the root directory and replace its contents with the following code.
+
+```javascript
+export default {
+  // Global page headers: https://go.nuxtjs.dev/config-head
+  head: {
+    title: "todolist-article",
+    htmlAttrs: {
+      lang: "en",
+    },
+    meta: [
+      { charset: "utf-8" },
+      { name: "viewport", content: "width=device-width, initial-scale=1" },
+      { hid: "description", name: "description", content: "" },
+      { name: "format-detection", content: "telephone=no" },
+    ],
+    link: [{ rel: "icon", type: "image/x-icon", href: "/favicon.ico" }],
+  },
+
+  // Global CSS: https://go.nuxtjs.dev/config-css
+  css: [],
+
+  // Plugins to run before rendering page: https://go.nuxtjs.dev/config-plugins
+  plugins: [],
+
+  // Auto import components: https://go.nuxtjs.dev/config-components
+  components: true,
+
+  // Modules for dev and build (recommended): https://go.nuxtjs.dev/config-modules
+  buildModules: [],
+
+  // Modules: https://go.nuxtjs.dev/config-modules
+  modules: [
+    // https://go.nuxtjs.dev/bootstrap
+    "bootstrap-vue/nuxt",
+    // Doc: https://axios.nuxtjs.org/usage
+    "@nuxtjs/axios",
+    "@nuxtjs/dotenv",
+    "@nuxtjs/auth-next",
+  ],
+
+  /*
+   ** Axios module configuration
+   ** See https://axios.nuxtjs.org/options
+   */
+  axios: {},
+
+  auth: {
+    strategies: {
+      okta: {
+        scheme: "openIDConnect",
+        endpoints: {
+          configuration:  `${process.env.OKTA_DOMAIN}/oauth2/default/.well-known/oauth-authorization-server`,
+          logout: undefined,
+        },
+        clientId: process.env.OKTA_CLIENT_ID,
+        grantType: "authorization_code",
+        responseType: "code",
+      },
+    },
+  },
+  // Build Configuration: https://go.nuxtjs.dev/config-build
+  build: {},
+};
+```
+
+The first thing this file does is make use of dotenv. It registers the modules you added earlier through the task runner and via npm. Finally, it sets up the options for your OAuth configuration. This is where you use the environment variables that you set up in your `.env` file.  
+
+You should note the line of code that reads `logout: undefined`. This is critical because for logging the user out correctly, Okta requires the `idToken` to be passed as a query parameter. However, `nuxt-auth` won't include that parameter under the hood. The solution is to override the logout URL obtained from the configuration endpoint with undefined and manually log the user out in your page file. You will implement this shortly.
```

Comment:
```suggestion
You should note the line of code that reads `logout: undefined`. It is critical for logging the user out correctly, because Okta requires the `idToken` to be passed as a query parameter. However, `nuxt-auth` won't include that parameter under the hood. The solution is to override the logout URL obtained from the configuration endpoint with `undefined` and manually log the user out in your page file. You will implement this shortly.
```
streamlined and added code syntax for clarity